source/common/ssl/context_impl.h

```diff
@@ -41,7 +42,8 @@ struct SslStats {
 
 class ContextImpl : public virtual Context {
 public:
-  virtual bssl::UniquePtr<SSL> newSsl() const;
+  virtual bssl::UniquePtr<SSL>
+  newSsl(const absl::optional<std::string>& override_server_name) const;
```

Comment:
Nit: making this argument an `absl::optional<absl::string_view>` (without reference) might be strictly more flexible.

Reply:
Done in 
https://github.com/envoyproxy/envoy/pull/4973/commits/505cce2d357b95c4caa7c15b3be4478e547c93ef